core/src/http/cached_resource_tests.rs

```diff
@@ -0,0 +1,292 @@
+use crate::{
+    http::cached_resource::{CachedResource, expires_at},
+    initialize_rustls,
+    retries::test_util::test_http_request_exponential_backoff,
+    test_util::install_test_trace_subscriber,
+};
+use http::{
+    HeaderValue,
+    header::{CACHE_CONTROL, CONTENT_TYPE},
+};
+use janus_messages::MediaType;
+use prio::codec::{Decode, Encode};
+use std::time::Duration;
+use tokio::time::Instant;
+use url::Url;
+
+#[derive(Debug, Eq)]
+struct TestResource {}
+
+impl Encode for TestResource {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), prio::codec::CodecError> {
+        ().encode(bytes)
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        ().encoded_len()
+    }
+}
+
+impl Decode for TestResource {
+    fn decode(_: &mut std::io::Cursor<&[u8]>) -> Result<Self, prio::codec::CodecError> {
+        Ok(TestResource {})
+    }
+}
+
+impl PartialEq for TestResource {
+    fn eq(&self, _: &Self) -> bool {
+        true
+    }
+}
+
+impl MediaType for TestResource {
+    const MEDIA_TYPE: &'static str = "application/test-resource";
+}
+
+#[tokio::test]
+async fn no_cache_control() {
+    install_test_trace_subscriber();
+    initialize_rustls();
+
+    tokio::time::pause();
+
+    let mut server = mockito::Server::new_async().await;
+    let server_url = Url::parse(&server.url()).unwrap();
+    let http_client = reqwest::Client::builder().build().unwrap();
+
+    let mock = server
+        .mock("GET", "/resource")
+        .with_status(200)
+        .with_body(TestResource {}.get_encoded().unwrap())
+        .with_header(CONTENT_TYPE.as_str(), TestResource::MEDIA_TYPE)
+        .expect(1)
+        .create_async()
+        .await;
+
+    // new() will fetch the resource from server. Because no cache-control is provided, resource()
+    // should refetch.
```

Comment:
This comment doesn't reflect the `expect(1)` above
```suggestion
    // new() will fetch the resource from server. Because no cache-control is provided, resource()
    // should not refetch.
```